mule-extensions-api-xml-dsl/src/main/java/org/mule/runtime/extension/xml/dsl/api/resolver/DslSyntaxResolver.java

```diff
@@ -286,6 +286,12 @@ public void visitArrayType(ArrayType arrayType) {
       }
 
       @Override
+      public void visitDictionary(DictionaryType dictionaryType) {
+        listBuilder.supportsChildDeclaration(false);
```

Comment:
don't hate me, but this shouldn't be validated here. If we reach this point is because the list actually supports child declaration. 

In `DslSyntaxUtils#supportsInlineDeclaration` we have a `visitArrayType` that only delegates in the generic type. We should change it there, so whenever an array has an `arrayType.getType()` that is dictionary we set the inline support to false.